digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/ManoeuvreService.scala

```diff
@@ -211,28 +218,130 @@ class ManoeuvreService(roadLinkService: RoadLinkService) {
+    }
+  }
+
+  def getSourceRoadLinkIdById(id: Long) : Long = {
     withDynTransaction {
       dao.getSourceRoadLinkIdById(id)
     }
   }
 
-  def find(id: Long) = {
+  def find(id: Long) : Option[Manoeuvre] = {
     withDynTransaction {
       dao.find(id)
     }
   }
 
+  private def createManoeuvreFromTrafficSign(manouvreProvider: ManoeuvreProvider, newTransaction: Boolean = true): Option[Long] = {
+    logger.info("creating manoeuvre from traffic sign")
+    val tsLinkId = manouvreProvider.trafficSign.linkId
+    val tsDirection = manouvreProvider.trafficSign.validityDirection
+
+    if (tsLinkId != manouvreProvider.sourceRoadLink.linkId)
+      throw new ManoeuvreCreationException(Set("Wrong roadlink"))
+
+    if (SideCode(tsDirection) == SideCode.BothDirections)
+      throw new ManoeuvreCreationException(Set("Isn't possible to create a manoeuvre based on a traffic sign with BothDirections"))
+
+    val connectionPoint = roadLinkService.getRoadLinkEndDirectionPoints(manouvreProvider.sourceRoadLink, Some(tsDirection)).headOption.getOrElse(throw new ManoeuvreCreationException(Set("Connection Point not valid")))
+
+    val (intermediates, adjacents, adjacentConnectPoint) = recursiveGetAdjacent(tsLinkId, connectionPoint, newTransaction)
+    val manoeuvreInit = manouvreProvider.sourceRoadLink +: intermediates
+
+    val roadLinks = getTrafficSignsProperties(manouvreProvider.trafficSign, "trafficSigns_type").map { prop =>
+      val tsType = TrafficSignType(prop.propertyValue.toInt)
+
+      tsType match {
+
+        case TrafficSignType.NoLeftTurn => manoeuvreInit :+ roadLinkService.pickLeftMost(manoeuvreInit.last, adjacents)
+
+        case TrafficSignType.NoRightTurn => manoeuvreInit :+ roadLinkService.pickRightMost(manoeuvreInit.last, adjacents)
+
+        case TrafficSignType.NoUTurn => {
+          val firstLeftMost = roadLinkService.pickLeftMost(manoeuvreInit.last, adjacents)
+          val (int, newAdjacents, _)= recursiveGetAdjacent(firstLeftMost.linkId, getOpositePoint(firstLeftMost.geometry, adjacentConnectPoint), newTransaction)
+          (manoeuvreInit :+ firstLeftMost) ++ (int :+ roadLinkService.pickLeftMost(firstLeftMost, newAdjacents))
+        }
+        case _ => Seq.empty[RoadLink]
+      }
+    }.getOrElse(Seq.empty[RoadLink])
+
+    if(!validateManoeuvre(manouvreProvider.sourceRoadLink.linkId, roadLinks.last.linkId, ElementTypes.FirstElement, newTransaction))
+      throw new ManoeuvreCreationException(Set("Manoeuvre creation not valid"))
+    else
+      Some(createWithoutTransaction("traffic_sign_generated", NewManoeuvre(Set(), Seq.empty[Int], None, roadLinks.map(_.linkId), Some(manouvreProvider.trafficSign.id)), roadLinks))
+
+  }
+
+  def createManoeuvreBasedOnTrafficSign(manouvreProvider: ManoeuvreProvider, newTransaction: Boolean = true): Option[Long] = {
+    if(newTransaction) {
+      withDynTransaction {
+        createManoeuvreFromTrafficSign(manouvreProvider, !newTransaction)
```

Comment:
is by default  newTransaction: Boolean = false we only need `createManoeuvreFromTrafficSign(manouvreProvider)`